torch/csrc/jit/fuser/cuda/resource_strings.h

```diff
@@ -85,11 +85,21 @@ constexpr auto rand_support_literal = R"(
       unsigned int nlo = (unsigned int)(n);
       unsigned int nhi = (unsigned int)(n >> 32);
       counter.x += nlo;
-      if (counter.x < nlo)
+      if (counter.x < nlo) {
+        unsigned int nhi_old = nhi;
         nhi++;
```

Comment:
There's no need to store `nhi_old`, from now on it's enough to check for `nhi==0` (that would mean that nhi overflown)

Reply:
Thank you! Changed `if(nhi_old <= nhi)` to `if(nhi != 0)`.